src/hotspot/cpu/aarch64/aarch64.ad

```diff
@@ -4421,6 +4432,28 @@ operand immI19()
   interface(CONST_INTER);
 %}
 
+// 5 bit signed integer
+operand immI5()
+%{
+  predicate(((-(1 << 4)) <= n->get_int()) && (n->get_int() < (1 << 4)));
```

Comment:
At some point someone must realize that

`((-(1 << size)) <= n->get_int()) && (n->get_int() < (1 << size))`

could be a function.

Reply:
> At some point someone must realize that
> 
> `((-(1 << size)) <= n->get_int()) && (n->get_int() < (1 << size))`
> 
> could be a function.

I think this function can make this predicate more clearly, but I found there are many predicates to modify in aarch64.ad.
I suggest using an extra patch to add this function and modify all similar predicates.